src/MainWindow.cpp

```diff
@@ -2437,59 +2426,82 @@ void MainWindow::reloadSettings()
+            if(newversion)
+            {
+                int ignmajor = Settings::getValue("checkversion", "ignmajor").toInt();
+                int ignminor = Settings::getValue("checkversion", "ignminor").toInt();
+                int ignpatch = Settings::getValue("checkversion", "ignpatch").toInt();
 
-        // check if the user doesn't care about the current update
-        if(!(ignmajor == major && ignminor == minor && ignpatch == patch))
-        {
-            QMessageBox msgBox;
-            QPushButton *idontcarebutton = msgBox.addButton(tr("Don't show again"), QMessageBox::ActionRole);
-            msgBox.addButton(QMessageBox::Ok);
-            msgBox.setTextFormat(Qt::RichText);
-            msgBox.setWindowTitle(tr("New version available."));
-            msgBox.setText(tr("A new DB Browser for SQLite version is available (%1.%2.%3).<br/><br/>"
-                              "Please download at <a href='%4'>%4</a>.").arg(major).arg(minor).arg(patch).
-                                arg(url));
-            msgBox.exec();
-
-            if(msgBox.clickedButton() == idontcarebutton)
+                // check if the user doesn't care about the current update
+                if(!(ignmajor == major && ignminor == minor && ignpatch == patch))
+                {
+                    QMessageBox msgBox;
+                    QPushButton *idontcarebutton = msgBox.addButton(tr("Don't show again"), QMessageBox::ActionRole);
+                    msgBox.addButton(QMessageBox::Ok);
+                    msgBox.setTextFormat(Qt::RichText);
+                    msgBox.setWindowTitle(tr("New version available."));
+                    msgBox.setText(tr("A new DB Browser for SQLite version is available (%1.%2.%3).<br/><br/>"
+                                    "Please download at <a href='%4'>%4</a>.").arg(major).arg(minor).arg(patch).
+                                        arg(url));
+                    msgBox.exec();
+
+                    if(msgBox.clickedButton() == idontcarebutton)
+                    {
+                        // save that the user don't want to get bothered about this update
+                        Settings::setValue("checkversion", "ignmajor", major);
+                        Settings::setValue("checkversion", "ignminor", minor);
+                        Settings::setValue("checkversion", "ignpatch", patch);
+                    }
+                }
+            }
+            else
             {
-                // save that the user don't want to get bothered about this update
-                Settings::setValue("checkversion", "ignmajor", major);
-                Settings::setValue("checkversion", "ignminor", minor);
-                Settings::setValue("checkversion", "ignpatch", patch);
+                if(!silent)
+                    QMessageBox::information(this, QApplication::applicationName(), tr("You are using the latest version."));
             }
         }
-    }
+    }, false, true);
```

Comment:
Not that I find this a problem, but probably the original implementation made the lambda function call another function so it didn't get so big. In fact, this final part seems to come from nowhere.

Reply:
Sorry, My knowledge is limited, so I don't quite understand. Could you explain it in a bit more detail? :pray:

Reply:
These two parameters are the last ones for the call to `RemoteNetwork::fetch()`:
https://github.com/sqlitebrowser/sqlitebrowser/blob/c6ce01b3c5702db28f79ea95a506b8eb2f5edd95/src/RemoteNetwork.h#L48

They come after the std::function parameter, which is passed to the call as a lambda function, that is, an anonymous, inline function. They're arguably too far from the beginning of the `fetch()` call. That could be solved reordering the parameters.

But moreover, Lambda functions are usually short, otherwise they seem convoluted against a call to a named function defined elsewhere.

Reply:
Understood, but changing the order of parameters could cause side effects,
so how about defining another function and calling it instead, as we originally did?

Reply:
Defining another function seems good to me.